api/nakadi-event-bus-api.yaml

```diff
@@ -799,6 +799,103 @@ paths:
             $ref: '#/definitions/Problem'
 
 
+  /subscriptions/{subscription_id}/offsets:
+    get:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Exposes the currently committed offsets of a subscription.
+      parameters:
+        - name: subscription_id
+          in: path
+          type: string
+          description: Id of subscription
+          required: true
+      responses:
+        '200':
+          description: Ok
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Cursor'
+        '204':
+          description: |
+            No Content. If the reading from the subscription was not started yet - there is nothing to expose.
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Client is not authorized to perform this operation
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Service (temporarily) unavailable
+          schema:
+            $ref: '#/definitions/Problem'
+    put:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Endpoint for committing offsets of the subscription. The offsets are per partition of EventType. One can use
+        this endpoint to commit the offsets for any of the partitions that are part of this subscription.
+        With every commit the offsets can only increase so it is not possible to commit offset that is lower than
```

Comment:
```
client A              Nakadi              client B

    +                    +                   +
    |POST initial Cursor |                   |
    +-------------------->                   |
    |OK Cursor V1        |                   |
    <--------------------+                   |
    |                    |GET Cursor V1      |
    |                    +------------------->
    |                    |                   |
    |PUT Cursor V1 update|                   |
    +-------------------->                   |
    |OK Cursor V2        |                   |
    <--------------------+                   |
    |                    |PUT Cursor V1 update
    |                    <-------------------+
    |                    |Conflict           |
    |                    +------------------->
    |                    |                   |
    |                    |                   |
    |                    |                   |
    +                    +                   +
```

This way, the first update to reach with a certain version "wins". Subsequent updates should be targeted to the most recent version.

But you are right about rewinding. We should have additional parameters for that, so let's ignore this by now.